script.js

```diff
@@ -1,4 +1,341 @@
+      } else if (cardName == 11) {
+        cardName = "jack";
+        // assigns score of 10 to each jack in the deck
+        blackJackScore = 10;
+      } else if (cardName == 12) {
+        // assigns score of 10 to each queen in the deck
+        cardName = "queen";
+        blackJackScore = 10;
+      } else if (cardName == 13) {
+        // assigns score of 10 to each king in the deck
+        cardName = "king";
+        blackJackScore = 10;
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: blackJackScore,
+      };
+
+      // Add the new card to the deck which was previously an empty array (see global variables at the top)
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+
+// a function that shuffles the deck we have
+var shuffleCards = function (cardDeck) {
+  // loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // select the card that corresponds to the random index
+    var randomCard = cardDeck[randomIndex];
+    // select the card that corresponds to the current index
+    var currentCard = cardDeck[currentIndex];
+    // swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // return the shuffled deck
+  return cardDeck;
+};
+
+// a function that deals two initial cards each to computer and player
+var initialDealRound = function () {
+  cardDeck = makeDeck();
+  shuffledDeck = shuffleCards(cardDeck);
+
+  // one card dealt each is first dealt to computer and player, then this repeats. so in total two cards each dealt to computer and player
+  // push 1st card from shuffled deck to computer's hand
+  computerHand.push(shuffledDeck.pop());
+  // push 1st card from shuffled deck to player's hand
+  playerHand.push(shuffledDeck.pop());
+  // push 2nd card from shuffled deck to computer's hand
+  computerHand.push(shuffledDeck.pop());
+  // push 2nd card from shuffled deck to player's hand
+  playerHand.push(shuffledDeck.pop());
+
+  // add up card scores to give computerTotalScore and playerTotalScore
+  // computerTotalScore = countHandScore(computerHand);
+  // playerTotalScore = countHandScore(playerHand);
+  computerTotalScore = computerHand[0].rank + computerHand[1].rank;
+  playerTotalScore = playerHand[0].rank + playerHand[1].rank;
+  console.log(`Computer's current total score: ${computerTotalScore}`);
+  console.log(`Player's current total score: ${playerTotalScore}`);
+
```

Comment:
great use of console.log . Some pseudo code are not needed as the names are self explanatory.

Example :
 // push 1st card from shuffled deck to computer's hand
  computerHand.push(shuffledDeck.pop());
  
  Remember to remove the unwanted code instead of leaving it a pseudo code